client/my-sites/checkout/composite-checkout/hooks/use-is-apple-pay-available.ts

```diff
@@ -0,0 +1,126 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useState } from 'react';
+import { LineItem } from '@automattic/composite-checkout';
+import debugFactory from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import { Stripe, StripeConfiguration } from 'calypso/lib/stripe';
+
+const debug = debugFactory( 'calypso:composite-checkout:use-is-apple-pay-available' );
+
+declare global {
+	interface Window {
+		ApplePaySession: ApplePaySessionInterface | undefined;
```

Comment:
This is splitting hairs: in situations where the key itself might not exist, I would prefer to use `ApplePaySession?: ApplePaySessionInterface;`